coralproject/talk-fe
# Reader flags on comments

## What does this PR do?

A few things, since there were small dependencies that were tricky to manage across PRs (trickiness caused by item 2 below):

1. Lets readers flag a comment.
2. Updates `talk-fe` to use relationships between items provided by shelf.
3. Adds default styling to `talk-fe`.
4. Adds accessibility to `talk-fe`.
5. Adds instructions to README.md for debugging backend config issues.

## How do I test this PR?

1. Start `coral-embed-stream` as described in readme.md.
2. You should be able to flag a comment. On doing so you should see a notification at the bottom of the screen.
3. Reload the page. The flagged comment should persist.
4. Activate a screen reader (cmd-F5 on a mac). Site should be accessible and a good user experience.
5. Buttons should be styled as per https://trello.com/c/Mk8Od2m4/65-create-default-reader-facing-styles
6. Tests should pass.

## Note

1. This PR upsets a default user if one is not found, all flags come from this user. The branch can be modified to create a unique user ID on each page load, but this will make it difficult to test that users’ flags persist across page loads.
2. As of this PR, some errors are still being created by this bug: https://github.com/coralproject/shelf/issues/299. Until this is resolved, comments may not be present on page reload unless they have replies.
